dvc/external_repo.py

```diff
@@ -45,12 +46,18 @@ def external_repo(
     config = _get_remote_config(url) if os.path.isdir(url) else {}
     config.update(cache_config)
 
-    root_dir = path if for_write else os.path.realpath(path)
+    if for_write:
+        root_dir = path
+        fs = None
```

Comment:
Probably we should get rid of `for_write`, it might already be broken.

Reply:
Yeah, it is only used in dvcx and it has been effectively abandoned :( But write-functionality (virtual, without real checked-out repo) would be great for gitfs in the future.